File: C#/0009.cs
public class Solution {
    public bool IsPalindrome(int x) {
        // Handle special cases
        if (x < 0 || (x % 10 == 0 && x != 0)) {
            return false;
        }
        
        int original = x;
        int reversed = 0;
        
        // Reverse the integer
        while (x > 0) {
            int digit = x % 10;
            reversed = reversed * 10 + digit;
            x = x / 10;
        }
        
        // Compare reversed with the original number
        return original == reversed;
    }
}

File: Rust/0009.rs
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        // Handle negative numbers and numbers ending in zero (except zero itself)
        if x < 0 || (x % 10 == 0 && x != 0) {
            return false;
        }
        
        let mut original = x;
        let mut reversed = 0;
        
        // Reverse the number
        while original > 0 {
            let digit = original % 10; // Get the last digit
            reversed = reversed * 10 + digit; // Build the reversed number
            original /= 10; // Remove the last digit from original
        }
        
        // Compare the reversed number with the original number
        x == reversed
    }
}
